src/main/kotlin/com/linchpino/ai/service/AIService.kt

```diff
@@ -0,0 +1,7 @@
+package com.linchpino.ai.service
+
+import com.linchpino.ai.model.RequestDetail
+
+interface AIService {
+    fun talkToAI(requestDetail: RequestDetail?): String?
```

Comment:
Do not propagate null into your software. make requestDetail a non nullable type